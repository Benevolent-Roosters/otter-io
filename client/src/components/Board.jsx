import React from 'react';
import { connect } from 'react-redux';
import { Carousel, Button, NavItem } from 'react-bootstrap';
import Slider from 'react-slick';
import PrevArrow from 'react-slick';
import '../../../ticketStyle.css';

import { getPanelsByBoard, getTicketsByPanel, toggleEditBoard, toggleCreatePanel } from '../redux/actionCreators.js';
import Panel from './Panel.jsx';
import CreateBoard from './CreateBoard.jsx';
import CreatePanel from './CreatePanel.jsx';
import CreateTicket from './CreateTicket.jsx';
import EditBoard from './EditBoard.jsx';
import EditPanel from './EditPanel.jsx';
import EditTicket from './EditTicket.jsx';

const Board = (props) => {
  let settings = {
    slidesToShow: 3,
    slidesToScroll: 1,
    speed: 500,
    infinite: false,
    centerMode: true,
    focusOnSelect: true,
    draggable: false,
    initialSlide: 3,
    useCSS: true,
    slickGoTo: 3
  };

  const panelStyle = {
    height: '650px'
  };

  const boardNameStyle = {
    fontSize: '20px',
    color: 'black',
    fontWeight: 'normal',
    display: 'flex',
    listStyle: 'none',
  };

  return (  
    <div>
      <CreateBoard />
      <CreatePanel />
      <CreateTicket />
      <EditBoard />
      <EditPanel />
      <EditTicket />
      <NavItem className="boardName" style={boardNameStyle} eventKey={1} onClick={() => {
        props.handleEditBoardRendered();}}>
        {props.currentBoard.board_name}
      </NavItem> 
      <Button onClick={() => props.handleCreatePanelRendered()}>Create Panel</Button>
      <Slider {...settings}>
        {props.panels.map(panel => 
        <div><Panel panelInfo={panel} key={panel.id} style={panelStyle}/></div>)}
      </Slider>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    'currentBoard': state.rootReducer.currentBoard,
    'panels': state.rootReducer.panels,
    currentPanel: state.rootReducer.currentPanel
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    handleGetPanelsByBoard(panels) {
      dispatch(getPanelsByBoard(panels));
    },
    handleGetTicketsByPanel(panelId) {
      dispatch(getTicketsByPanel(panelId));
    },
    handleEditBoardRendered() {
      dispatch(toggleEditBoard());
    },
    handleCreatePanelRendered() {
      dispatch(toggleCreatePanel());
    }
  };
};

export var UnwrappedBoard = Board;
export default connect(mapStateToProps, mapDispatchToProps)(Board);

