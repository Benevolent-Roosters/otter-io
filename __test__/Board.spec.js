import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Board, { UnwrappedBoard } from '../client/src/components/Board.jsx';
import { Provider } from 'react-redux';
import rootReducer from '../client/src/redux/reducers.js';

describe('<Board />', () => {
  let store = {
    currentBoard: {
      boardId: 0,
      teamName: '',
      repoName: '',
      owner: '',
      repoUrl: ''
    },
    panels: []
  };
  it ('should render the unconnected Board correctly', () => {
    const wrapper = shallow(<UnwrappedBoard />);
    expect(wrapper).toMatchSnapshot();
  });
  it ('should render the connected Board', () => {
    const wrapper = shallow(<Provider store={store}><Board /></Provider>);
    expect(wrapper).toMatchSnapshot();
  })
})