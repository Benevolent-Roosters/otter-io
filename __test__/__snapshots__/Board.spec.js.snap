// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> should render the connected Board 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(Board) />,
  "nodes": Array [
    <Connect(Board) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "currentBoard": Object {
                          "boardId": 0,
                          "owner": "",
                          "repoName": "",
                          "repoUrl": "",
                          "teamName": "",
                        },
                        "panels": Array [],
                      }
        }
>
        <Connect(Board) />
</Provider>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(Board) />,
          "store": Object {
            "currentBoard": Object {
              "boardId": 0,
              "owner": "",
              "repoName": "",
              "repoUrl": "",
              "teamName": "",
            },
            "panels": Array [],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "currentBoard": Object {
            "boardId": 0,
            "owner": "",
            "repoName": "",
            "repoUrl": "",
            "teamName": "",
          },
          "panels": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(Board) />,
        "_debugID": 4,
        "_renderedOutput": <Connect(Board) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "currentBoard": Object {
              "boardId": 0,
              "owner": "",
              "repoName": "",
              "repoUrl": "",
              "teamName": "",
            },
            "panels": Array [],
          }
    }
>
    <Connect(Board) />
</Provider>,
}
`;

exports[`<Board /> should render the unconnected Board correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    I love pie
</div>,
  "nodes": Array [
    <div>
      I love pie
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Board />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          I love pie
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          I love pie
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Board />,
}
`;
